wqflask/wqflask/correlation/pre_computes.py

```diff
@@ -167,3 +168,49 @@ def get_datasets_data(base_dataset, target_dataset_data):
         samples_fetched, base_traits_data)
 
     return (target_results, base_results)
+
+
+def fetch_text_file(dataset_name, conn, text_dir=TEXTDIR):
+    """fetch textfiles with strain vals if exists"""
+
+    with conn.cursor() as cursor:
+        query = 'SELECT Id, FullName FROM ProbeSetFreeze WHERE Name = "%s"' % dataset_name
+        cursor.execute(query)
+        results = cursor.fetchone()
+    if (results):
+        try:
+            for file in os.listdir(text_dir):
+                if file.startswith(f"ProbeSetFreezeId_{results[0]}_"):
+                    return os.path.join(text_dir, file)
+        except FileNotFoundError:
+            pass
+
+
+def read_text_file(sample_dict, file_path):
+
+    def parse_line_csv(line):
+        return_list = line.split('","')
+        return_list[-1] = return_list[-1][:-2]
+        return_list[0] = return_list[0][1:]
+        return return_list
+
+    def filter_line_with_index(line, index):
+        lst = parse_line_csv(line)
+        return ",".join([lst[i] for i in index])
+
+    def __fetch_id_positions__(all_ids, target_ids):
+        _vals = []
+        _posit = [0]  # alternative for parsing
+
+        for (idx, strain) in enumerate(all_ids, 1):
+            if strain in target_ids:
+                _vals.append(target_ids[strain])
+                _posit.append(idx)
+
+        return (_posit, _vals)
+    with open(file_path, "r") as file_handler:
+        all_ids = file_handler.readline()
+        _posit, sample_vals = __fetch_id_positions__(
+            parse_line_csv(all_ids)[1:], sample_dict)
+        return (sample_vals, [filter_line_with_index(line, _posit)
+                              for line in file_handler.readlines()])
```

Comment:
**issue:** As @fredmanglis mentioned, use python's inbuilt csv module to parse csv files.